packages/shared/src/components/modals/SubmitArticle.tsx

```diff
@@ -0,0 +1,211 @@
+import React, { ReactElement, useContext, useRef, useState } from 'react';
+import { useMutation } from 'react-query';
+import request from 'graphql-request';
+import { Button } from '../buttons/Button';
+import { ModalCloseButton } from './ModalCloseButton';
+import { ModalProps } from './StyledModal';
+import { formToJson } from '../../lib/form';
+import { apiUrl } from '../../lib/config';
+import {
+  contentGuidlines,
+  requestAccessSubmitArticle,
+} from '../../lib/constants';
+import { ResponsiveModal } from './ResponsiveModal';
+import { TextField } from '../fields/TextField';
+import LinkIcon from '../../../icons/link.svg';
+import { SUBMIT_ARTICLE_MUTATION } from '../../graphql/submitArticle';
+import PostItemCard from '../post/PostItemCard';
+import { PostItem } from '../../graphql/posts';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+
+type SubmitArticleProps = {
+  isEnabled: boolean;
+  headerCopy: string;
+  submitArticleModalButton: string;
+} & ModalProps;
+
+export default function SubmitArticle({
+  isEnabled,
+  headerCopy,
+  submitArticleModalButton,
+  onRequestClose,
+  ...modalProps
+}: SubmitArticleProps): ReactElement {
+  const submitFormRef = useRef<HTMLFormElement>();
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [enableSubmission, setEnableSubmission] = useState(false);
+  const [isValidating, setIsValidating] = useState(false);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [existingArticle, setExistingArticle] = useState<PostItem>(null);
+  const [urlHint, setUrlHint] = useState<string>();
+
+  const { mutateAsync: submitArticle } = useMutation<unknown, unknown, string>(
+    (articleUrl: string) =>
+      request(`${apiUrl}/graphql`, SUBMIT_ARTICLE_MUTATION, {
+        url: articleUrl,
+      }),
+  );
+
+  const submitArticleFailEvent = (reason: string): void => {
+    trackEvent({
+      event_name: 'submit article failed',
+      extra: JSON.stringify({
+        reason,
+      }),
+    });
+  };
+
+  const onUrlChanged = () => {
+    if (submitFormRef.current) {
+      if (existingArticle) {
+        setExistingArticle(null);
+      }
+      if (urlHint) {
+        setUrlHint(null);
+      }
+      setEnableSubmission(submitFormRef.current.checkValidity());
+    }
+  };
+
+  const onSubmit = async (
+    event: React.FormEvent<HTMLFormElement>,
+  ): Promise<void> => {
+    event.preventDefault();
+
+    trackEvent({
```

Comment:
Let's add the actual URL as extra please

Reply:
Just checking, you wanted to track every click right, or only ones where the URL is definitely set?

Reply:
You can see the change proposed now :)